### ISSUE TITLE
Support for labels in Angular expressions

### ISSUE THREAD

**Issue by btford** [angular/angular#5445]
After chatting this morning with @alxhub @matsko @yjbanov and @vsavkin, we agreed to add labels to the Angular expression language to support #2845.

The end result will allow us to write something like:

```html
<a [router-link]="link: /User(id: some.bound.value)">Link to User 5</a>
```

Which will be transformed to:

```html
<a [router-link]="['/User', {id: some.bound.value}]">Link to User 5</a>
```

The important bit for this is that our expression parser accepts something like `([a-z]+):(\w+)` and provides the bit after the colon as a string that can be transformed.

**Comment by ewinslow** [comment]
I realize you haven't written much about the design here, but to me it seems that the new way suffers a couple key problems:

 * it is actually longer to type than the old
 * it uses more Angular custom syntax
   * people have to learn the quirks of this DSL
   * you have to write tooling for it

**Comment by btford** [comment]
@ewinslow see #2845 for discussion

**Comment by ewinslow** [comment]
I read it. What is the intended takeaway?

**Comment by btford** [comment]
Ah, your points are valid, but we should continue discussion there.

**Comment by otodockal** [comment]
One problem what I have with this proposal is that it doesn't look like JavaScript. 
I know DSL is DSL :), but in expression I suppose to write JS. Also transformed part (former) is better to remember.

**Comment by btford** [comment]
@otodockal your point is valid, but can we please continue discussion on #2845? Thanks.

**Comment by btford** [comment]
Closed via #5452.
